### DISCUSSION TITLE
Suggest：nnUNet can be designed to run as a server，so as to improve performance

### DISCUSSION THREAD

**Discussion by goodsave** [MIC-DKFZ/nnUNet#1673] [Ideas]
If nnUNet can be designed to run as a server, keep Pytorch waiting after loading the model, stay in memory, and use a queue to process the received segmentation tasks, this will greatly improve performance and avoid wasting time caused by repeated model loading.

**Comment by ancestor-mithril** [comment]
I don't think this is feasible, because nnUNet is designed to predict using an ensemble of models trained on different folds. As a consequence, each model is loaded (sequentially) to do inference on a single input file. 
If the models are not loaded, all models will have to be kept in memory.

**Comment by goodsave** [comment]
Okay, thank you very much for your reply.
